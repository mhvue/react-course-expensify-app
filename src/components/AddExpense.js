import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";

export class AddExpense extends React.Component {
    onSubmit= (expense) => {
        this.props.startAddExpense(expense)
        this.props.history.push("/dashboard");
    }

    render() {
        return (
            <div>
            <h1>Add Expense</h1>
            <ExpenseForm 
            onSubmit={this.onSubmit}
                />
        </div>
        );
    }
}

//which lets you create functions that dispatch when called, and pass those functions as props to your component.
const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => {
        dispatch(startAddExpense(expense))
    }
})

export default connect(undefined, mapDispatchToProps)(AddExpense);
