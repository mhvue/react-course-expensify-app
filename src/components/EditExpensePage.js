import React from 'react';
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from '../actions/expenses';
import { connect } from "react-redux";

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
    
            this.props.editExpense(this.props.expense.id, expense);
            this.props.history.push('/');
    }

    onRemove = () => {
        this.props.removeExpense({id: this.props.expense.id});
        this.props.history.push("/")
        }

    render() {
        return(
        <div>
         <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Edit Expense</h1>
            </div>
        </div>

         <div className="content-container"> 
                <ExpenseForm 
                expense={this.props.expense}
                onSubmit = {this.onSubmit}
                />
                <button 
                onClick={this.onRemove}
                className="button button--secondary"> 
                Remove Expense</button> 
         </div>
         </div>
        )
    }
};


const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
    }
};

const mapDispatchToProps = (dispatch, props) => {
    return {
        editExpense: (id,expense) => {
            dispatch(editExpense(id,expense))
        },
        removeExpense: (data) => {
            dispatch(removeExpense(data))
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);