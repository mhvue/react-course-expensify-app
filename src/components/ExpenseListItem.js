import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseItemList = ({id, description, createdAt, amount}) =>  (
    <div>
    <Link to={`/edit/${id}`}>   
      <p>{description}</p>
    </Link>
      <p>Created at: {moment(createdAt).format("MMMM Do, YYYY")} </p>
       <p>Amount: {numeral(amount / 100).format("$0,0.00")}</p>     
    </div>
);


export default ExpenseItemList;