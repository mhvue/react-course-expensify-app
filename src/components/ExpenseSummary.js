import React from "react";
import { connect } from "react-redux";
import selectExpenseTotal from "../selectors/expense-total";
import selectExpenses from "../selectors/expenses";
import numeral from 'numeral';

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    
    return (
      <div>
        <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
      </div>
    );
  };


const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpense.length,
        expensesTotal: selectExpenseTotal(visibleExpense)
    }
}
export default connect(mapStateToProps)(ExpenseSummary);