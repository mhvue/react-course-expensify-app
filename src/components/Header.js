import React from 'react';
import {NavLink} from "react-router-dom";

//want header on every single page 
//activeClassName is a class that gets added to the link when we are on it 
const Header= () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName= "is-active" exact={true}>Dashbaord</NavLink>
        <NavLink to="/create" activeClassName= "is-active">Create Expense</NavLink>
    </header>
);

<<<<<<< HEAD

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined,mapDispatchToProps)(Header);
=======
export default Header;
 
>>>>>>> parent of 5d067fd (added in authentication from firebase. added testing for logging in and out.)
