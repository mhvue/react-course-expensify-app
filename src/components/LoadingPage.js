import React from "react";

const LoadingPage = () => (
    <div className="loader">
        Loading
        <img className="loader__image" src="/images/loader.gif"/>
    </div>
)

export default LoadingPage;