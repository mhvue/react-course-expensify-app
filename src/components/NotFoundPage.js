import React from 'react';
import {Link} from "react-router-dom";


//using Link is preferred for internal componenets to take advantage of client side routing 
const NotFoundPage = () => (
    <div>
    404 - <Link to="/">Go Home</Link> 
    </div>
);

export default NotFoundPage;