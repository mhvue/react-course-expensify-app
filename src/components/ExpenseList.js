import React from "react";
import { connect } from "react-redux"; //connect your component to the redux store 
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";

//as the store changes, our componenet will also update 
export const ExpenseList = (props) =>  (
    <div className="content-container">
    <div className="list-header">
    <div className="show-for-mobile">Expenses</div>
    <div className="show-for-desktop">Expense</div>
    <div className="show-for-desktop">Amount</div>
    </div>

    <div className="list-body">
        {
            props.expenses.length === 0 ? (
                <div className="list-item list-item--message">
                    <span>No expenses</span>
                </div>
            ): (
                props.expenses.map((expense) => {
                return <ExpenseListItem 
                    key = {expense.id}
                    {...expense}
                    />
            })
            )
            
        }
    </div>
    </div>
);

//as the store changes, below will get rerun 
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
}

//inside connect, we want to pass a function saying waht information we want to access from store 
export default connect(mapStateToProps)(ExpenseList);

