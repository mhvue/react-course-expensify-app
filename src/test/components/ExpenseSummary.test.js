import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import { expect } from "@jest/globals";

test("should render ExpenseSummary",() => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with multiple expenses",() => {
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={45235}/>)
    expect(wrapper).toMatchSnapshot();
})