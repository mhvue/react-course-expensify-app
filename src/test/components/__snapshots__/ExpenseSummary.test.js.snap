// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $452.35
  </h1>
</div>
`;
